onnxruntime/python/tools/transformers/fusion_utils.py

```diff
@@ -123,6 +124,24 @@ def get_dtype(self, shape_infer_helper, input_or_output_name: str) -> int:
 
         return None
 
+    def remove_cascaded_cast_nodes(self):
+        """Remove Cast node that are overrided by another Cast node like  --> Cast --> Cast -->
+           Note that this shall be used carefully since it might introduce semantic change.
+           For example, float -> int -> float could get different value than the original float value.
```

Comment:
I see this is getting called only once and first part of this comment is taken care of. 
Is the second part 
'float -> int -> float could get different value than the original float value.' taken care somewhere else?

Reply:
Mixed precision conversion adds some "Cast" nodes to the graph. Here we have a post processing to remove some "Cast" nodes that is not needed.